examples/flutter_firebase_login/packages/authentication_repository/lib/src/authentication_repository.dart

```diff
@@ -89,6 +96,55 @@ class AuthenticationRepository {
     }
   }
 
+  /// Starts the Sign In with Apple Flow.
+  ///
+  /// Throws a [logInWithApple] if an exception occurs.
+  Future<void> logInWithApple() async {
+    // To prevent replay attacks with the credential returned from Apple, we
+    // include a nonce in the credential request. When signing in with
+    // Firebase, the nonce in the id token returned by Apple, is expected to
+    // match the sha256 hash of `rawNonce`.
+    final rawNonce = generateNonce();
+    final nonce = sha256ofString(rawNonce);
+
+    try {
+      final appleCredential = await SignInWithApple.getAppleIDCredential(
+        scopes: [
+          AppleIDAuthorizationScopes.email,
+          AppleIDAuthorizationScopes.fullName,
+        ],
+        nonce: nonce,
+      );
+
+      // Create an `OAuthCredential` from the credential returned by Apple.
+      final credential = firebase_auth.OAuthProvider('apple.com').credential(
+        idToken: appleCredential.identityToken,
+        rawNonce: rawNonce,
+      );
+
+      await _firebaseAuth.signInWithCredential(credential);
+    } on Exception {
+      throw LogInWithAppleFailure();
+    }
+  }
+
+  /// Generates a cryptographically secure random nonce, to be included in a
+  /// credential request.
+  String generateNonce([int length = 32]) {
```

Comment:
```suggestion
  String _generateNonce([int length = 32]) {
```
Thought it could be better if we keep it private as we don't use it anywhere else. What do you think?

Reply:
moved to `crypto.dart`.